core/eth/contracts/FluxAggregator.go

```diff
@@ -0,0 +1,181 @@
+package contracts
+
+import (
+	"math/big"
+
+	"chainlink/core/eth"
+	"chainlink/core/store/models"
+	"chainlink/core/utils"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	"github.com/shopspring/decimal"
+)
+
+//go:generate mockery -name FluxAggregator -output ../../internal/mocks/ -case=underscore
+
+type FluxAggregator interface {
+	eth.ConnectedContract
+	SubscribeToLogs(fromBlock *big.Int) (LogSubscription, error)
+	LatestAnswer(precision int32) (decimal.Decimal, error)
+	LatestRound() (*big.Int, error)
+	ReportingRound() (*big.Int, error)
+	TimedOutStatus(round *big.Int) (bool, error)
+	LatestSubmission(oracleAddress common.Address) (*big.Int, *big.Int, error)
+}
+
```

Comment:
I'm game for waiting for review to implement documentation, but it will definitely be needed at some point

Reply:
Yeah, I'm all for that.